beacon_node/eth1/src/http.rs

```diff
@@ -423,6 +439,29 @@ fn response_result(response: &str) -> Result<Option<Value>, String> {
     }
 }
 
+/// Extracts the human-readable message string from a RPC error.
+///
+/// Note: the input string, `s`, will be of the form:
+///     "Eth1 node returned error: {\"code\":[code],\"message\":\"[message]\"}"
+fn rpc_error_msg(s: &str) -> Result<String, String> {
```

Comment:
Working on the error string here by manually splitting the json string seems a little convoluted. Instead you can directly work on the json object by modifying the `response_result` function to return the error string itself. Something like
```rust
/// Accepts an entire HTTP body (as a string) and returns either the `result` field or the `error['message']` field, as a serde `Value` .
fn response_result_or_error(response: &str) -> Result<Result<Value, Value>, String> {
    let json = serde_json::from_str::<Value>(&response)
        .map_err(|e| format!("Failed to parse response: {:?}", e))?;

    if let Some(error) = json.get("error").map(|e| e.get("message")).flatten() {
        Ok(Err(error.clone()))
    } else {
        let res = json
            .get("result")
            .cloned()
            .ok_or_else(|| "Failed to get result".to_string())?;
        Ok(Ok(res))
    }
}
```

Then in the `get_chain_id` function, you can check the error string from the above function.

Reply:
This is much more elegant, provided we're happy with modifying the other handlers to accommodate the changed function signature?

Reply:
Actually I think this is a pretty clean change, consult `e77a41a`.